service/matching/db.go

```diff
@@ -58,8 +58,10 @@ type (
 		subqueues []persistencespb.SubqueueInfo
 	}
 
+	subqueueKey int
```

Comment:
sorry if this is too annoying...

I think this should be
```suggestion
	subqueueIndex int
```
we already have a concept called "subqueue key": the proto message that differentiates the subqueues (currently only has priority key but we can put extra splitting stuff there)

Reply:
It's a good call 👍